### DISCUSSION TITLE
Building An OSS Kontakt Template

### DISCUSSION THREAD

**Discussion by Xykrotik** [pumodi/open-samples#8] [General]
I am in/will be in the process of building/creating an easy to use, minimal Kontakt Template for use on OSS. 
This will speedup the time it takes for us to go from the editing and post-processing stage of a project to the published library available for download. 

This template will have documentation on what is already setup in the template library, how to navigate the Kontakt UI and backend for adding samples, groups, etc, and more. The goal is to cover as much as possible to make it easier for other people to contribute their own libraries to the project. The documentation will be updated as things are added/removed (New Kontakt Versions, new parameters, etc) as well as with additional details for anything that may not be clear or may become a frequently asked question. 

While the licensing of the template is not yet created, for now the idea is that the template's licensing will restrict usage to **only** Open Source Samples projects, as we do not want it to be used in commercial sample libraries or for monetary gain. Plus, it gives all of OSS' Kontakt sample libraries a consistent look, which while not necessary, is nice! 

You as a contributor will still be able to create your own libraries from scratch without the template, using your own design choices, UI, etc. and are more than welcome to contribute those to the repository. You may also modify the OSS template to fit your own design choices, UI, etc, it does NOT have to match the default template at all, it's just a tool to help make it easier for you to make a library! However, even when modifying the OSS template, it is still restricted to usage with OSS projects only please! 

That's all for now, I'll update further as things progress. 

~Xykrotik

**Comment by pumodi** [comment]
That is a brilliant idea. Love it! Let me know how I can assist!

**Comment by Xykrotik** [comment]
**### October 4th, 2022 marks the official release of Kontakt 7 from Native Instruments.** 

Kontakt 7, as all newer Kontakt versions have been, is backwards compatible with previous versions of Kontakt libraries. 

That said, the Open Source Samples Kontakt Template will be created in Kontakt 6, as not everyone wants to, has the funds to, or cares to upgrade to Kontakt 7. This means that if you are not on Kontakt 6 by now, unfortunately, the Kontakt version of libraries using the template will not be compatible with Kontakt versions older than 6 (_full version number pending decision_). We will not be updating anything to Kontakt 7 until the release of Kontakt 8, as we do not want to limit availability anytime soon. That said, if you desire to make, **in addition to a K6 version**, a K7 version of your own or an existing OSS library, you are welcome to do so, however a K6 version of the library **_must_** exist. (If you own Kontakt, you have the ability to download older version of the plugin directly from Native Instruments)

Thankfully, as is the goal with the project, even if you are unable to use the Kontakt version of the library, we will have a multitude of other formats as well as RAW samples for you to utilize for your music projects, that way no one gets excluded from using what they want! 

I'll continue to update this post as things move along, but wanted to address the stance on Kontakt Dev at this time, considering it is a very widely used plugin and most likely will be the most used version of our libraries. 

Thanks!
Xykrotik

**Reply by pumodi** [reply]
I'm all for this! Great writeup
